src/librustc_interface/passes.rs

```diff
@@ -1102,7 +1103,12 @@ pub fn start_codegen<'tcx>(
 
     // Output Yorick debug sections into binary targets.
     if tcx.sess.crate_types.borrow().contains(&config::CrateType::Executable) {
-        let (def_ids, _) = tcx.collect_and_partition_mono_items(LOCAL_CRATE);
+        let def_ids = tcx.collect_and_partition_mono_items(LOCAL_CRATE).0;
+        let def_ids: &DefIdSet = def_ids.borrow();
```

Comment:
This binding is needed to avoid:
```
error[E0282]: type annotations needed
    --> src/librustc_interface/passes.rs:1107:54
     |
1107 |         let mut def_ids: DefIdSet = def_ids.borrow().clone::<DefIdSet>();
     |                                                      ^^^^^ cannot infer type for `Borrowed`
     |
     = note: type must be known at this point
```

I'd like to get rid of the intermediate binding if possible, but all of my attempts so far have failed.

Any ideas how to fix?

Reply:
I suspect there's an implicit `deref` or `as_ref` or similar being added to this so that this line really expands to something like:

```rust
let def_ids: &DefIdSet = def_ids.borrow().deref();
```

because otherwise the `Borrowed` type that `borrow()` returns wouldn't be compatible with `DefIdSet`.

Reply:
So shall we leave it as is?

Reply:
Well, if you can change it to one line, it would probably be easier to read, and you'll know for next time, but it's up to you.

Reply:
I'm confused because the set is behind an `Arc` whose borrow is implemented like this:
```
#[stable(feature = "rust1", since = "1.0.0")]
impl<T: ?Sized> borrow::Borrow<T> for Arc<T> {
    fn borrow(&self) -> &T {
        &**self
    }
}
```

The type parameter `Borrowed` is only visible in the trait def:
```
pub trait Borrow<Borrowed: ?Sized> {
    fn borrow(&self) -> &Borrowed;
```

Reply:
Ah, OK, then I don't know what the type inference error you encountered was!

Reply:
Apparently it's a clash of the clones (sounds like a star wars film). Fix coming.